Extensions/BBText/bbtextruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * The PIXI.js renderer for the BBCode Text runtime object.
+ *
+ * @class BBTextRuntimeObjectPixiRenderer
+ * @constructor
+ * @param {gdjs.BBTextRuntimeObject} runtimeObject The object to render
+ * @param {gdjs.RuntimeScene} runtimeScene The gdjs.RuntimeScene in which the object is
+ */
+gdjs.BBTextRuntimeObjectPixiRenderer = function(runtimeObject, runtimeScene) {
+  this._object = runtimeObject;
+
+  // Load (or reset) the text
+  if (this._pixiObject === undefined) {
+    this._pixiObject = new MultiStyleText(runtimeObject._text, {
+      default: {
+        fontFamily: runtimeObject._fontFamily,
+        fontSize: `${runtimeObject._fontSize}px`,
+        fill: runtimeObject._color,
+        tagStyle: 'bbcode',
+        wordWrap: runtimeObject._wordWrap,
+        wordWrapWidth: runtimeObject._wrappingWidth,
+        align: runtimeObject._align,
+      },
+    });
+
+    this._object.hidden = !runtimeObject._visible;
+  } else {
+    this.updateText();
+    this.updateColor();
+    this.updateAlignment();
+    this.updateFontFamily();
+    this.updateFontSize();
+    this.updatePosition();
+  }
+
+  runtimeScene
+    .getLayer('')
+    .getRenderer()
+    .addRendererObject(this._pixiObject, runtimeObject.getZOrder());
+
+  // Set the anchor in the center, so that the object rotates around
+  // its center
+  this._pixiObject.anchor.x = 0.5;
+  this._pixiObject.anchor.y = 0.5;
+
+  this.updateText();
+  this.updatePosition();
+  this.updateAngle();
+  this.updateOpacity();
+  this.updateVisible();
+  this.ensureUpToDate();
+};
+
+gdjs.BBTextRuntimeObjectRenderer = gdjs.BBTextRuntimeObjectPixiRenderer;
+
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.getRendererObject = function() {
+  return this._pixiObject;
+};
+
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.ensureUpToDate = function() {
+  if (this._pixiObject._style.wordWrap !== this._object._wordWrap) {
```

Comment:
What make wordWrap and wrappingWidth special compared to text/color/alignment/font size/opacity so that they need to be checked in ensureUpToDate?

Can you just call updateWordWrap and updateWrapping whenever they are changed from the object?

Reply:
They affect the size (width/height) of the instance and that has an effect on its positioning. The wrappingWidth is driven by the width of the instance, set at the beginning. If you don't change them via an action, or you don't set the text of the object via a command, without these checks during update, the instance is initiated at the wrong position. So this check here sort of resolves that

Reply:
In theory, calling `setWrappingWidth` in `extraInitializationFromInitialInstance ` should call internally setPosition and so the positioning should be right.
If that's not the case, it's that there is something that modify the object/renderer without calling the proper updates.

Might be related to the texture, you can try to remove these and put calls to updateWrappingWidth and updateWordWrap inside something that is launched a single time ```js
gdjs.BBTextRuntimeObjectPixiRenderer = function(runtimeObject, runtimeScene) {
  // ...
  // At the end:
  this._firstRendering = true;
}

gdjs.BBTextRuntimeObjectPixiRenderer.prototype.ensureUpToDate = function() {
  if (this._firstRendering) {
    // Call whatever is needed.
    this._firstRendering = false;
  }
}
```

so that you just pay for one boolean check at every frame, which is slightly better than comparing two strings.

Reply:
ok, turns out that I can remove these